tests/components/smtp/test_notify.py

```diff
@@ -85,3 +91,51 @@ def test_html_email(self, mock_make_msgid):
             "Test msg", data={"html": html, "images": ["test.jpg"]}
         )
         assert "Content-Type: multipart/related" in msg
+
+
+async def test_reload_notify(hass):
+    """Verify we can reload the notify service."""
+
+    with patch(
+        "homeassistant.components.smtp.notify.MailNotificationService.connection_is_valid"
+    ):
+        assert await async_setup_component(
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6103615)